src/index.ts

```diff
@@ -249,7 +249,7 @@ export const localStorageSync = (config: LocalStorageConfig) => (
       if (state) {
         Object.keys(state).forEach(function (key) {
           if (state[key] instanceof Array && rehydratedState[key] instanceof Array) {
-              state[key] = rehydratedState[key];
+              state = {...state, key: rehydratedState[key]};
```

Comment:
if we have key=foo
afaik 
 _state[key] = 'bar'_ will generate a {foo:'bar'}
while this line will generate _{key:'bar'}_
use {...state, [key]: rehydratedState[key]} instead

![image](https://user-images.githubusercontent.com/32337426/51088226-9b879300-1765-11e9-912a-2ee33ffaab48.png)